game.js

```diff
@@ -454,7 +453,18 @@ class Game {
         const winloss = (this.state.winner === this.conn.bot_id ? "W" : "L");
         this.log(`Game over.   Result: ${col}+${res}  ${winloss}`);
 
-        if (this.bot) {
+        // Notify bot of end of game and send score
+        if (config.farewellscore && this.bot) {
+            const sendTheScore = (score) => {
+                if (score) this.log(`Bot thinks the score was ${score}`);
+                if (res !== "R" && res !== "Time" && res !== "Can") this.sendChat(`Final score was ${score} according to the bot.`, "discussion");
```

Comment:
Why this check? It wasn't required in the last line, and also why not check for score like in the previous line?

Shouldn't the check for send chat and log be equal?

Reply:
Also i do not think res can change here, so we could check for res before we send the bot the score command and skip it all together in that case.

Reply:
Also i do not think res can change here, so we could check for res before we send the bot the score command and skip it all together in that case.

Reply:
Also i do not think res can change here, so we could check for res before we send the bot the score command and skip it all together in that case.

Reply:
Also i do not think res can change here, so we could check for res before we send the bot the score command and skip it all together in that case.

Reply:
resign, timeout or game cancelled.
as I said, my major motivation for this was a signal that the game has ended, the chat is kind of an aside.

Reply:
So if you do not send the chat message for resign, cancel or timeout, what remains where you do send it?

Reply:
double pass, where score actually makes sense. but in all cases the bot knows the game has ended, and can save it's own log, which is frankly the main thing I am using this for. Happy to rename it, and/or remove the chat.

Reply:
1. I think it's interesting to show final score on resign too. In fact on double pass, ogs score is usually accurate.
2. is it possible to detect and whitelist double pass instead of excluding time and cancel? Seems safer to use a whitelist for the things we want to show, instead of blacklist some. That way we also wont act on unknown values in case ogs adds something new or we forgot an option.

Reply:
1. done . 2. not easy and I think it's not too bad if we send on some unknown value.